src/history/hs.c

```diff
@@ -975,6 +977,14 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
                 continue;
             }
 
+            /* Clear history store content if we skip inserting the updates without timestamp. */
+            if (clear_hs && upd->start_ts != WT_TS_NONE) {
+                /* We can only delete history store entries that have timestamps. */
```

Comment:
Why?

Reply:
In a case that U2@20 -> U1@0 and U2 is globally visible, we skip inserting U1 which is a mixed mode operation. We need to clear the history store before U2 is inserted.